### PR TITLE
[build-presets] Support expansion of preset option names

### PR DESCRIPTION
**Pull request by ahoppen**
This allows me to selectively enable build features (such as building lldb, building swiftpm reconfiguring etc.) locally from a single preset.

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please smoke test

File: utils/build_swift/build_swift/presets.py [hunk 1]
```diff
@@ -361,11 +361,23 @@ def _get_preset(self, name):
     def _interpolate_preset_vars(self, preset, vars):
         interpolated_options = []
         for (name, value) in preset.options:
-            try:
-                value = _interpolate_string(value, vars)
-            except KeyError as e:
-                raise InterpolationError(
-                    preset.name, name, value, e.args[0])
+            # If the option is a key-value pair, e.g. 
+            # install-destdir=%(install_dir)s
```
Comment (gottesmm) [inline utils/build_swift/build_swift/presets.py:365]: Can you add a test for this.

Comment (ahoppen) [reply utils/build_swift/build_swift/presets.py:365]: Test case added.

**Comment by ahoppen** [comment]
@swift-ci Please smoke test

**Comment by ahoppen** [comment]
@swift-ci Please smoke test Windows

**Comment by ahoppen** [comment]
@gottesmm Could you take another look at the PR?

**Review by gottesmm — APPROVED** [review]

**Pull request merged**